src/main/scala/cromwell/CromwellCommandLine.scala

```diff
@@ -0,0 +1,104 @@
+package cromwell
+
+import java.nio.file.{Files, Path, Paths}
+
+import better.files._
+import cromwell.core.{ErrorOr, WorkflowSourceFiles}
+import cromwell.util.FileUtil._
+import lenthall.exception.MessageAggregation
+
+import scalaz._
+import Scalaz._
+import scala.util.{Failure, Success, Try}
+
+sealed abstract class CromwellCommandLine
+case object UsageAndExit extends CromwellCommandLine
+case object RunServer extends CromwellCommandLine
+final case class RunSingle(wdlPath: Path,
+                           sourceFiles: WorkflowSourceFiles,
+                           inputsPath: Option[Path],
+                           optionsPath: Option[Path],
+                           metadataPath: Option[Path]) extends CromwellCommandLine
+
+object CromwellCommandLine {
+  def apply(args: Seq[String]): CromwellCommandLine = {
+    args.headOption match {
+      case Some("server") if args.size == 1 => RunServer
+      case Some("run") if args.size >= 2 && args.size <= 5 => RunSingle(args.tail)
+      case _ => UsageAndExit
+    }
+  }
+}
+
+object RunSingle {
+  def apply(args: Seq[String]): RunSingle = {
+    val wdlPath = Paths.get(args.head)
+    val inputsPath = argPath(args, 1, Option(".inputs"), checkDefaultExists = false)
+    val optionsPath = argPath(args, 2, Option(".options"), checkDefaultExists = true)
```

Comment:
Thinking about this during lunch (at commonwealth, what is this, a weekday?) I started to get more concerned about the "something subtle going on" aspect. 

Questions like this and your other comment regarding defaults & such make me think the right path forward would be to sit down at some point and decide how we think things *should* work and then make the code/docs more clear. Perhaps the former is a no op once we work it through, but the latter definitely isn't. I'll make a ticket and advocate it as part of the 0.23 milestone